dockerfiles/Dockerfile.readiness

```diff
@@ -0,0 +1,37 @@
+FROM quay.io/projectquay/golang:1.25 AS builder
+ARG TARGETOS
+ARG TARGETARCH
+
+# Install build tools
+# The builder is based on UBI8, so we need epel-release-8.
+RUN dnf install -y 'https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.noarch.rpm' && \
+    dnf install -y gcc-c++ libstdc++ libstdc++-devel clang zeromq-devel pkgconfig && \
+    dnf clean all
+
+WORKDIR /workspace
+
+# Copy the Go Modules manifests
+COPY go.mod go.mod
+COPY go.sum go.sum
+
+# Copy the go source
+COPY cmd/ cmd/
+COPY pkg/ pkg/
+
+# Build
+
+ENV CGO_ENABLED=1
+ENV GOOS=${TARGETOS:-linux}
+ENV GOARCH=${TARGETARCH}
+ARG BUILD_REF
+RUN go build -a -o bin/readiness cmd/readiness/main.go
+
+# Use ubi9 as a minimal base image
+# Refer to https://catalog.redhat.com/software/containers/ubi9/ubi-minimal/615bd9b4075b022acc111bf5 for more details
+FROM registry.access.redhat.com/ubi9/ubi-minimal:latest
+WORKDIR /
+COPY --from=builder /workspace/bin/readiness /app/readiness
+
+USER 65532:65532
```

Comment:
IIRC, this gives OpenShift indigestion (i.e., requires special privileges and handling). IIRC, the easy path with OpenShift is to not specify the userid and groupid in the image, because OpenShift wants to assign those.